src/shared/tpm2-util.c

```diff
@@ -3635,6 +3516,212 @@ static int tpm2_build_sealing_policy(
         return 0;
 }
 
+#if HAVE_OPENSSL
+static int tpm2_ecc_curve_from_openssl_curve_id(int curve_id, TPM2_ECC_CURVE *ret) {
+        assert(ret);
+
+        switch (curve_id) {
+        case NID_X9_62_prime192v1: *ret = TPM2_ECC_NIST_P192; return 0;
+        case NID_secp224r1:        *ret = TPM2_ECC_NIST_P192; return 0;
+        case NID_X9_62_prime256v1: *ret = TPM2_ECC_NIST_P256; return 0;
+        case NID_secp384r1:        *ret = TPM2_ECC_NIST_P384; return 0;
+        case NID_secp521r1:        *ret = TPM2_ECC_NIST_P521; return 0;
+        case NID_sm2:              *ret = TPM2_ECC_SM2_P256;  return 0;
```

Comment:
that's a strange way of indentation, how we never do it otherwise.

maybe encode this in a static const array? 

```c
static const struct {
        int openssl_curve_id;
        TPM2_ECC_CURVE tpm2_ecc_curve;
} table[] = {
        { NID_X9_62_prime192v1, TPM2_ECC_NIST_P192 },
        …
};

FOREACH_ARRAY(t, table, ELEMENTSOF(table)) 
        if (t->openssl_curve_id == curve_id)
                 return t->tpm2_ecc_curve;
…
```

Reply:
ack